__tests__/integration/utils/benchmark/tx-benchmark.util.js

```diff
@@ -0,0 +1,57 @@
+import { loggers } from '../logger.util';
+import { delay } from '../core.util';
+
+/**
+ * @typedef TxInstance
+ * @property {string} type
+ * @property {string} [txId]
+ * @property {number} [startTime]
+ * @property {number} [endTime]
+ * @property {number} [duration]
+ */
+
+/**
+ * List of tx instances that were started
+ * @type TxInstance[]
+ */
+const instances = [];
+
+export class TxBenchmarkUtil {
+  /**
+   * Adds a transaction instance to the local storage
+   * @param {TxInstance} txData
+   */
+  static addTx(txData) {
+    instances.push({ ...txData });
+  }
+
+  /**
+   * Calculates a summary of transaction length for all instances available locally.
+   * @returns {{}}
+   */
+  static calculateSummary() {
+    const summary = {};
+    let sumTxDuration = 0;
+
+    for (const tx of instances) {
+      sumTxDuration += tx.duration;
+    }
+    summary.sumTxDuration = sumTxDuration;
+    summary.amountTxs = (instances.length + 1);
+    summary.avgTxDuration = sumTxDuration / summary.amountTxs;
+
+    return summary;
+  }
+
+  /**
+   * A method to write all the local storage instances into a log file via Winston.
+   * @returns {Promise<void>}
+   */
+  static async logResults() {
+    for (const txIndex in instances) {
+      const metadata = { tx: instances[txIndex] };
+      loggers.txBenchmark.insertLineToLog('Tx instance', metadata);
+      await delay(0); // Necessary to allow each log request to be fulfilled
```

Comment:
Winston seems to be unable to output the log in the speed the code requires because it runs on the same single-thread as the whole application. There are some issues opened about this, the most descriptive I found being [1364](https://github.com/winstonjs/winston/issues/1364) that is still open after 4 years.

The best solution to a production system would be to open a separate process dedicated to transporting to filesystem, which would just receive the messages and work the pooling/flushing to `fs`, as the PinoJS logging lib also recommends.

For the purposes of just flushing a test result, I found this solution reasonable ( inspired by [a SO answer](https://stackoverflow.com/a/31543718/1123007) ). An alternative would be to bypass winston and use `fs.appendToFile` directly.